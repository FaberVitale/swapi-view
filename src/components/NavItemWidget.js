//@flow
import React from "react";
import NavigationLink from "./NavigationLink";
import BackButton from "./BackButton";

type Props = {
  route: string
};

/* Navigation widget used in Item */
class NavItemWidget extends React.Component<Props> {
  render() {
    let to, label;

    if (/\/item\//.test(this.props.route)) {
      to = this.props.route.replace("item", "wookiee");
      label = "Wookiee";
    } else {
      to = this.props.route.replace("wookiee", "item");
      label = "Item";
    }

    return (
      <div className="flex-row align-items-center">
        <BackButton />
        <NavigationLink to={to} disabled={false} label={label} replace={true} />
      </div>
    );
  }
}

export default NavItemWidget;
