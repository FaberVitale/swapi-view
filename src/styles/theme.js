//@flow
import {
  yellow200 as primary1,
  yellow500 as primary2,
  grey400 as accent3
} from "material-ui/styles/colors";
import darkBaseTheme from "material-ui/styles/baseThemes/darkBaseTheme";
import getMuiTheme from "material-ui/styles/getMuiTheme";

const boxShadow =
  "0 1px 6px rgba(0, 0, 0, 0.12), 0 1px 4px rgba(0, 0, 0, 0.12)";

const theme = getMuiTheme(darkBaseTheme, {
  palette: {
    primary1Color: primary1,
    primary2Color: primary2,
    accent3Color: accent3,
    accent1Color: primary1,
    alternateTextColor: darkBaseTheme.palette.canvasColor
  },
  topBar: {
    backgroundColor: darkBaseTheme.palette.canvasColor,
    padding: 0,
    margin: 0,
    boxShadow: boxShadow
  },

  paper: {
    boxShadow
  },

  homePaper: {
    backgroundColor: "#000000"
  },

  extLink: {
    color: primary1,
    outlineColor: primary2
  },

  h1: {
    color: primary2,
    fontSize: 24
  },
  h2: {
    color: primary2
  },

  noResults: {
    color: "rgba(255, 255, 255, .5)"
  },

  autocomplete: {
    icon: {
      fill: primary1
    }
  },

  textField: {
    floatingLabelColor: "#fefefe",
    focusColor: primary2,
    borderColor: "#fefefe",
    hintColor: "rgba(255, 255, 255, .5)"
  },

  dropDownMenu: {
    accentColor: "#fefefe",
    focusColor: primary2
  },

  h3: darkBaseTheme.h4,

  p: {
    color: primary2
  },
  failedToLoad: {
    frown: {
      color: primary2,
      opacity: 0.1
    }
  },
  menuItem: {
    selectedTextColor: primary2
  }
});

export default theme;
