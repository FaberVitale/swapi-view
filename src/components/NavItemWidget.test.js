import React from "react";
import Renderer from "react-test-renderer";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import theme from "../styles/theme";
import NavItemWidget from "./NavItemWidget";
import { MemoryRouter } from "react-router-dom";

describe("components/NavItemWidget", () => {
  const item = Renderer.create(
    <MemoryRouter initialEntries={["/"]} iniitalIndex={0}>
      <MuiThemeProvider muiTheme={theme}>
        <NavItemWidget route="/people/item/5" />
      </MuiThemeProvider>
    </MemoryRouter>
  ).toJSON();

  const wookie = Renderer.create(
    <MemoryRouter initialEntries={["/"]} iniitalIndex={0}>
      <MuiThemeProvider muiTheme={theme}>
        <NavItemWidget route="/people/wookiee/5" />
      </MuiThemeProvider>
    </MemoryRouter>
  );

  it("matches the snapshots", () => {
    expect(item).toMatchSnapshot();
    expect(wookie).toMatchSnapshot();
  });
});
