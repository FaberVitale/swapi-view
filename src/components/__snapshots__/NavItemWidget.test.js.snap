// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/NavItemWidget matches the snapshots 1`] = `
<div
  className="flex-row align-items-center item-nav"
>
  <button
    aria-label="go back"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
  </button>
  <a
    className="pointer-events-auto"
    disabled={false}
    href="/people/wookiee/5"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "minWidth": 88,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        style={
          Object {
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0,
            "muiPrepared": true,
            "paddingLeft": 16,
            "paddingRight": 16,
            "position": "relative",
            "textTransform": "uppercase",
            "verticalAlign": "middle",
          }
        }
      >
        Wookiee
      </span>
    </div>
  </a>
</div>
`;

exports[`components/NavItemWidget matches the snapshots 2`] = `
<div
  className="flex-row align-items-center item-nav"
>
  <button
    aria-label="go back"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button"
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(255, 255, 255, 1)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
  </button>
  <a
    className="pointer-events-auto"
    disabled={false}
    href="/people/item/5"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "backgroundColor": "rgba(0, 0, 0, 0)",
        "border": 10,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "rgba(255, 255, 255, 1)",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": "inherit",
        "fontWeight": "inherit",
        "height": 36,
        "lineHeight": "36px",
        "margin": 0,
        "minWidth": 88,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "hidden",
        "padding": 0,
        "position": "relative",
        "textAlign": "center",
        "textDecoration": "none",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
        "verticalAlign": null,
      }
    }
    tabIndex={0}
  >
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      <span
        style={
          Object {
            "fontSize": 14,
            "fontWeight": 500,
            "letterSpacing": 0,
            "muiPrepared": true,
            "paddingLeft": 16,
            "paddingRight": 16,
            "position": "relative",
            "textTransform": "uppercase",
            "verticalAlign": "middle",
          }
        }
      >
        Item
      </span>
    </div>
  </a>
</div>
`;
